lib/Common/Memory/RecyclerWriteBarrierManager.cpp

```diff
@@ -30,6 +33,7 @@ BYTE* RecyclerWriteBarrierManager::cardTable = RecyclerWriteBarrierManager::x64C
 #else
 // Each byte in the card table covers 4096 bytes so the range covered by the table is 4GB
 BYTE RecyclerWriteBarrierManager::cardTable[1 * 1024 * 1024];
+bool dummy = RecyclerWriteBarrierManager::Initialize();
```

Comment:
I was looking at this too; I read that it was to initialize the "g_..." function for the other "CopyArray" code to verify "no barrier needed" when the other no barrier variants were used. But this looks heavy weighted and incur penalty for general case -- looks the other code always call this "g_..." function no matter what. I think you should be able to (1) only do the verification in debug/test builds, and (2) forward declare a global function instead of creating this global variable.

---
In reply to: [86642266](https://github.com/Microsoft/ChakraCore/pull/1920#discussion_r86642266) [](ancestors = 86642266)